web/src/redux/persistent/projects/connections/reducer.ts

```diff
@@ -10,16 +10,28 @@ import {
 } from './actions'
 import { CREATE_OUTCOME_WITH_CONNECTION, DELETE_OUTCOME_FULLY } from '../outcomes/actions'
 import { isCrud, crudReducer } from '../../crudRedux'
+import { CellIdString, HeaderHashB64 } from '../../../../types/shared'
+import { Connection, CreateOutcomeWithConnectionOutput, DeleteOutcomeFullyResponse } from '../../../../types'
 
-const defaultState = {}
-// how to define an interface where the keys are variable?
 
+type State = { 
+  [cellId: CellIdString]: {
+    [headerHashOrPreviewId: string]: {
+      parentAddress: HeaderHashB64,
+      childAddress: HeaderHashB64,
+      randomizer: number, //i64,
+      isImported: boolean,
+      headerHash?: HeaderHashB64 // unable to do spread operation in type definition
```

Comment:
For this we can use a typescript "intersection" modifier, kind of similar to the union operator, but good for this situation.

https://www.typescriptlang.org/docs/handbook/2/objects.html#intersection-types



E.g. 

Connection & { headerHash: HeaderHashB64 }